pkg/detector/detector.go

```diff
@@ -284,8 +284,15 @@ func (d *ResourceDetector) EventFilter(obj interface{}) bool {
 		return false
 	}
 
-	if strings.HasPrefix(clusterWideKey.Namespace, names.KubernetesReservedNSPrefix) ||
-		strings.HasPrefix(clusterWideKey.Namespace, names.KarmadaReservedNSPrefix) {
+	reservedNamespace := map[string]struct{}{
+		names.NamespaceKarmadaCluster: {},
+		names.NamespaceKarmadaSystem:  {},
+		names.NamespaceKubeSystem:     {},
+		names.NamespaceKubePublic:     {},
+		names.NamespaceKubeNodeLease:  {},
+	}
```

Comment:
I guess we can add a common function at `pkg/util/names` package, thus we don't need to initialize it in each loop.